oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistrationRepositoryUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.registration;
+
+import org.springframework.util.Assert;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Utility methods for InMemoryClientRegistrationRepository and InMemoryReactiveClientRegistrationRepository.
+ * 
+ * @author David Perez
+ * @since 5.1
+ * @see InMemoryClientRegistrationRepository
+ * @see InMemoryReactiveClientRegistrationRepository
+ */
+public final class ClientRegistrationRepositoryUtils {
+
+	private ClientRegistrationRepositoryUtils() {
+	}
+
+	static Map<String, ClientRegistration> toUnmodifiableMapByRegistrationId(Stream<ClientRegistration> registrations) {
```

Comment:
I'd prefer to keep the input as a Collection as it is one less method invocation for the consumer of this API

Reply:
I'd like to see this renamed to `createClientRegistrationIdToClientRegistration`. This aligns more with Spring Security's naming conventions.

Reply:
ok